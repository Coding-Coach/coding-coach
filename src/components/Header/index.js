import React from 'react';

import logoImage from '../../assets/images/logo.svg';

export default React.memo(function Header({ children, title, subtitle }) {
  return (
    <header className="bg-primary-light bg-diagonal bg-diagonal-primary-light">
      <div className="container pt-20 pb-10 flex">
        <div className="flex-1">
          <h1 className="font-display font-semibold text-white text-7xl uppercase tracking-wider leading-tight md:whitespace-pre">{title}</h1>
          {subtitle && <p className="text-white text-sans text-2xl mb-4 font-body font-hairline leading-tight">{subtitle}</p>}
          {children}
        </div>
        <div className="hidden md:block flex-1">
          <img src={logoImage} className="w-5/6 float-right" alt="The coding coach logo" />
        </div>
      </div>
    </header>
  );
});
