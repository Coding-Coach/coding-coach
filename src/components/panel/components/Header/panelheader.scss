@import '~theme/base/variables';

.heading {
  margin: 0 auto 10px;
  text-transform: uppercase;
  color: $color-primary;
  font-family: $font-secondary;
  font-weight: $font-regular;
  font-size: $font-xl;
}