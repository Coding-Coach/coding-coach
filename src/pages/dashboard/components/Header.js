import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Trans } from '@lingui/macro';
import logoSrc from './assets/logo.svg';

function TopHeader() {
  return (
    <div className={classNames('bg-gradient-primary h-48 py-5', 'md:py-6 md:h-64 lg:py-8')}>
      <div className="flex justify-between items-center w-5/6 m-auto">
        <span className={classNames('text-white text-2xl font-bold')}>
          <Trans id="dashboard.header.title" />
        </span>
        <img src={logoSrc} className="w-12" />
      </div>
    </div>
  );
}

function Header({ banner }) {
  return (
    <React.Fragment>
      <header>
        {/* This includes the Section name and the Logo of CodingCoach */}
        <TopHeader />
      </header>

      {banner && (
        <div
          style={{ marginTop: '-6rem' }}
          className={classNames('mx-auto w-5/6 p-8', 'bg-white shadow', 'md:w-3/5 xl:w-2/5')}
        >
          {banner}
        </div>
      )}
    </React.Fragment>
  );
}

Header.propTypes = {
  banner: PropTypes.node,
};

export default Header;
