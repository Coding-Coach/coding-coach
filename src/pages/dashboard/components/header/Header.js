import React from 'react';
import PropTypes from 'prop-types';
import { Trans, t } from '@lingui/macro';
import { I18n } from '@lingui/react';
import IconLogo from 'components/icon/CCLogo';

function Header({ banner }) {
  return (
    <React.Fragment>
      <header>
        {/* This includes the Section name and the Logo of CodingCoach */}
        <TopHeader />
      </header>

      {banner && (
        <div
          style={{ marginTop: '-6rem' }}
          className="mx-auto w-5/6 p-8 bg-white shadow md:w-3/5 xl:w-2/5"
        >
          {banner}
        </div>
      )}
    </React.Fragment>
  );
}

function TopHeader() {
  return (
    <div className="bg-gradient-primary h-48 py-5 md:py-6 md:h-64 lg:py-8">
      <div className="flex justify-between items-center w-5/6 m-auto">
        <span className="text-white text-2xl font-bold md:text-4xl">
          <Trans id="dashboard.header.title" />
        </span>
        <I18n>
          {({ i18n }) => (
            <IconLogo
              className="w-16 md:w-24"
              alt={i18n._(t('home.header.logo')`This is the logo for Coding Coach`)}
            />
          )}
        </I18n>
      </div>
    </div>
  );
}

Header.propTypes = {
  banner: PropTypes.node,
};

export default Header;
