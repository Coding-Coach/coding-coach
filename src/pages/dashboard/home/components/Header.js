import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Trans, t } from '@lingui/macro';
import { I18n } from '@lingui/react';
import logoSrc from './assets/logo.svg';

function TopHeader() {
  return (
    <div className={classNames('bg-gradient-primary h-48 p-5', 'md:p-6 md:h-64 lg:p-8')}>
      <div className="flex justify-between items-center">
        <span className={classNames('text-white text-2xl font-bold')}>
          <Trans id="dashboard.header.title" />
        </span>
        <img src={logoSrc} className="w-12" />
      </div>
    </div>
  );
}
function Header({ banner }) {
  return (
    <header>
      <TopHeader />

      {banner && (
        <div
          className={classNames('bg-white p-8 shadow')}
          style={{ width: '80%', margin: '-100px auto 0 auto' }}
        >
          {banner}
        </div>
      )}
    </header>
  );
}

Header.propTypes = {
  banner: PropTypes.node,
};

export default Header;
