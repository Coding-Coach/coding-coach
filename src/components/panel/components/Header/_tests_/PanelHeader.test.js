import React from 'react';
import { render } from 'react-testing-library';
import Header from '../PanelHeader';

test('renders the PanelHeader component', () => {
  const { container } = render(<Header />);

  expect(container.firstChild).toMatchSnapshot();

  const h2 = container.querySelectorAll('h2');
  expect(h2.length).toEqual(1);
});
