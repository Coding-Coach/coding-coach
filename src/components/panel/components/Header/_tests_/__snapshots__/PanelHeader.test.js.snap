// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the PanelHeader component 1`] = `
<h2
  class="heading"
/>
`;
